### ISSUE TITLE
sbt.global.base doesn't work properly with '~'

### ISSUE THREAD

**Issue by erikerlandson** [sbt/sbt#4298]
- [x ] I've read the [CONTRIBUTING](https://github.com/sbt/sbt/blob/1.x/CONTRIBUTING.md) guidelines

## steps
```sh
$ sbt -Dsbt.global.base=~/.sbt/foo
```
## problem

global sbt file(s) under `~/.sbt/foo` are not read

## expectation

sbt should read the global information under `~/.sbt/foo`

## notes

sbt version: 1.2.0

workaround: use fully qualified home directory path, e.g. `/home/user/.sbt/foo`

**Comment by eed3si9n** [comment]
`~` is a feature of Bash - https://www.gnu.org/software/bash/manual/html_node/Tilde-Expansion.html

Maybe it's friendlier if sbt expanded it for you, but I'd say it's more of a nice-to-have feature than a bug.

**Comment by eed3si9n** [comment]
Having said that, this might be fairly easy to add, so if @erikerlandson or someone else to contribute to it, this might be a good first issue.

The relevant code is https://github.com/sbt/sbt/blob/v1.2.1/main/src/main/scala/sbt/BuildPaths.scala#L89-L92

```scala
  def getFileProperty(name: String): Option[File] = Option(System.getProperty(name)) flatMap {
    path =>
      if (path.isEmpty) None else Some(new File(path))
  }
```

That gets used by `getGlobalBase`.

**Comment by jvican** [comment]
Last time I looked at this you could not used a relative path as sbt global base, either, it had to be absolute.

**Comment by erikerlandson** [comment]
possibly I just read too much into some other docs.  I've seen paths like `~/.sbt/1.0/plugins/gpg.sbt`, and assumed `~` was allowed, but wasn't in the immediate context of setting `sbt.global.base`

Thanks for explaining!

**Comment by kai-chi** [comment]
this can be closed - fixed with #4367

**Comment by erikerlandson** [comment]
That is great! I'll close this.
